pkg/didcomm/packager/packager.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package packager
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/envelope"
+)
+
+// Provider contains dependencies for the base packager and is typically created by using aries.Context()
+type Provider interface {
+	envelope.Provider
+	Packers() []envelope.Packer
+	DefaultPacker() envelope.Packer
+}
+
+// BasePackager is the basic implementation of Packager
+type BasePackager struct {
```

Comment:
Is there a need for the prefix `Base` ?

Reply:
No, since it's not injectable we don't expect others.

Reply:
Then again there is the distinction between this and the interface (the interface is `Packager`), since we also have a mock packager. The interface is what everything consumes.

I could make the struct definition private.

Reply:
The interface and struct shouldn't be in the same package.

Reply:
I can return the interface instead of a pointer to the struct

Reply:
Where should I move the interface? It's consumed all over the codebase, particularly in tests which usually use the mock packager.

Reply:
I've resolved this by pulling the `Packager` interface out into `pkg/didcomm/common/transport`